s3: add FileLu S5 (S3-Compatible Object Storage) provider support

kingston125 opened the pull request:
#### What is the purpose of this change?


Add support for FileLu S5, an S3-compatible object storage service.
This includes a new provider block in backend/s3.go, updates to the provider list in the documentation, and adds region options for us-east, eu-central, ap-southeast, and me-central.


#### Was the change discussed in an issue or in the forum before?


Not previously discussed in an issue. This PR introduces initial support directly.


#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

kingston125 commented:
Currently the Encoding test is still failing. I am working on it.

ncw commented:
Not all the providers rclone supports pass the encoding test so failing that test is not so bad.

When you say you are working on it, are you working on the FileLu code or the rclone code?

kingston125 commented:
@ncw > Not all the providers rclone supports pass the encoding test so failing that test is not so bad.

All the upload, download, tag, metadata, create, delete, bucket, and object functions are working perfectly. If it’s okay even though the encoding test failed, can you please merge this?

kingston125 commented:
I accidentally closed this PR. I will fix the conflict.

kingston125 commented:
> Sorry we merged a fix putting the AWS config back into alphabetical order which has caused a conflict here :-( Can you fix the conflict @kingston125 then I'll merge this - thank you.

I have resolved the conflict and placed it in the correct alphabetical order.

Thanks @ncw

roucc commented:
> Thank you for the reply. Can you please specify which line that is?

```
Error: Autogenerated region of file modified at docs/content/s3.md line 841
Error: Autogenerated region of file modified at docs/content/s3.md line 842
Error: Autogenerated region of file modified at docs/content/s3.md line 875
Error: Autogenerated region of file modified at docs/content/s3.md line 876
Error: Autogenerated region of file modified at docs/content/s3.md line 1307
Error: Autogenerated region of file modified at docs/content/s3.md line 1310
```
Can be seen here https://github.com/rclone/rclone/pull/8836/files

kingston125 commented:
> > Thank you for the reply. Can you please specify which line that is?
> 
> ```
> Error: Autogenerated region of file modified at docs/content/s3.md line 841
> Error: Autogenerated region of file modified at docs/content/s3.md line 842
> Error: Autogenerated region of file modified at docs/content/s3.md line 875
> Error: Autogenerated region of file modified at docs/content/s3.md line 876
> Error: Autogenerated region of file modified at docs/content/s3.md line 1307
> Error: Autogenerated region of file modified at docs/content/s3.md line 1310
> ```
> 
> Can be seen here https://github.com/rclone/rclone/pull/8836/files

So can you please help me and tell me which file I should edit, s3.go or s3.md? Is s3.md auto-generated? I added FileLu S3 manually.

kingston125 commented:
> > So can you please help me and tell me which file I should edit, s3.go or s3.md? Is s3.md auto-generated? I added FileLu S3 manually.
> 
> Confusingly, only some of s3.md is auto-generated. Just remove the edits in the auto-generated regions (see linter output for exactly where) and leave the other changes.

Thank you @ncw and @roucc  , I have updated the s3.md
